docs/AddNewProduct.md

```diff
@@ -0,0 +1,96 @@
+# Adding a New Firebase Product Apple SDK
+
+## Introduction
+
+This document provides guidance on many of the factors to consider when designing and developing
+a new Firebase Product Apple SDK. The list is not totally comprehensive and there is ongoing
+evolution, so you should also consult with the iCore team as you are ramping up.
+
+## Swift
+
+While much of Firebase has been implemented in Objective C, any new products or major
+implementations should be implemented in Swift. If Objective C API support is required it should
+be implemented via the Swift `@objc` attribute. Consult with iCore and product management about
+the requirement itself.
+
+Apple and others provide many great guides for Swift programming. Googlers, see also this [2022
+presentation](go/swift-sdks-2022).
+
+Existing Firebase Swift implementations can be helpful. However, note that they are mostly
+Objective C ports and do not take advantage of Swift features like structs, default arguments, and
+async/await as much as new implementations should.
+
+### Swift Style
+
+Follow this [Style Guide](https://google.github.io/swift/).
+
+Firebase uses [swiftformat](https://github.com/nicklockwood/SwiftFormat) for enforcing code
+formatting consistency.
+
+## Package Managers
+
+Firebase supports four different distributions - Swift Package Manager, CocoaPods, Carthage, and
+binary zip distributions.
+
+It can be developed via Swift Package Manager or CocoaPods.
+
+The new project should set up a `podspec` for CocoaPods and add a `product` specification to
+[Package.swift](Package.swift).
+
+## Testing
+
+All Firebase code should be unit tested. Ideally, the unit tests should be driven from both
+CocoaPods and Swift Package Manager. If only one is implemented, it should be SPM.
+
+## Dependencies
+
+Dependencies are a great way to add libraries of well-tested functionality to Firebase. On the flip
+side, they can add bloat and risk to the Firebase user experience. Ideally, only existing Firebase
+dependencies should be used. See [Package.swift](Package.swift). If a new dependency is needed,
+consider making it a weak dependency and only hard depending on a protocol. Any new hard
```

Comment:
I don't quite follow the _and only hard depending on a protocol_ part. Could you maybe rework the back half of that sentence?

Reply:
@ncooke3 PTAL